Domas: document unified alerting rule list view (#34667)
grafana/grafana: docs/sources/alerting/unified-alerting/rule-list.md

+++
title = "View alert rules"
description = "View alert rules"
keywords = ["grafana", "alerting", "guide", "rules", "view"]
weight = 400
+++

# View alert rules

> In the Grafana side bar, hover your cursor over the Alerting (bell) icon and then click **Alert Rules**. By default it will list all configured Grafana alert rules as well as any rules from Loki or Prometheus data sources. By default the group view is shown. You can toggle between group or state views by clicking relevant **View as** buttons in the options area the top of the page.

Reviewer: ```suggestion
1. In the Grafana menu hover your cursor over the Alerting (bell) icon.
1. Click **Alert Rules**. You can see all configured Grafana alert rules as well as any rules from Loki or Prometheus data sources. 
By default, the group view is shown. You can toggle between group or state views by clicking the relevant **View as** buttons in the options area at the top of the page.
```

> ## Group view

Reviewer: ```suggestion
### Group view
```

> ![Grouped alert rule view](/img/docs/alerting/unified/rule-list-group-view-8-0.png 'Screenshot of grouped alert rule view')

Reviewer: Move the image to after description.

> Group view shows Grafana alert rules grouped by folder and Loki or Prometheus alert rules grouped by namespace + group. This is the default rule list view, intended to be used for managing rules. You can expand each group to view a list of rules in this group. Each rule can be expanded to view it's details, action buttons and any alerts spawned by this rule, and each alert can be further expanded to view it's details.
> 

Reviewer: ```suggestion
Group view shows Grafana alert rules grouped by folder and Loki or Prometheus alert rules grouped by `namespace` + `group`. This is the default rule list view, intended for managing rules. You can expand each group to view a list of rules in this group. Each rule can be further expanded to view its details. Action buttons and any alerts spawned by this rule, and each alert can be further expanded to view its details.

```

> ## State view

Reviewer: ```suggestion
### State view
```

> ![Alert rule state view](/img/docs/alerting/unified/rule-list-state-view-8-0.png 'Screenshot of alert rule state view')

Reviewer: Move the image to after description.

> State view shows alert rules grouped by state. Use this view to get an overview of which rules are in what state. Each rule can be expanded to view it's details, action buttons and any alerts spawned by this rule, and each alert can be further expanded to view it's details.

Reviewer: ```suggestion
State view shows alert rules grouped by state. Use this view to get an overview of which rules are in what state. Each rule can be expanded to view its details. Action buttons and any alerts spawned by this rule, and each alert can be further expanded to view its details.
```

## Filter alert rules

> You can use filter controls to view only alert rules that match specific criteria:

Reviewer: ```suggestion
You can use the following filters to view only alert rules that match specific criteria:
```

- **Filter alerts by name or label -** Type an alert name, label name or value in the **Search** input.

> - **Filter alerts by state -** In **States**, select which alert states you want to see. All others will be hidden.

Reviewer: ```suggestion
- **Filter alerts by state -** In **States** Select which alert states you want to see. All others are hidden.
```

> - **Filter alerts by datasource -** Click the **Select datasource** and select an alerting datasource. Only alert rules that query selected datasource will be visible.

Reviewer: "datasource" should be the "data source". It should be changed in the UI as well.

## Rule details

> ![Alert rule details](/img/docs/alerting/unified/rule-details-8-0.png 'Screenshot of alert rule details')

Reviewer: Again, move the image.

> A rule row will show rule state, health and summary annotation the rule has one. Rule row can be expanded to reveal rule labels, all annotations, datasources this rule queries and list of alerts spawned from this rule.

Reviewer: ```suggestion
A rule row shows the rule state, health, and summary annotation if the rule has one. You can expand the rule row to display rule labels, all annotations, data sources this rule queries, and a list of alerts spawned from this rule.
```

### Rule state

Rule state can be one of the following:

> - **Normal -** rule evaluation result is negative and it is not firing.
> - **Pending -** rule evaluation result is positive and it will become firing after **for** period passes.
> - **Firing -** rule evaluation result is positive and it is spawning alerts. 
> 

Reviewer: ```suggestion
- **Normal:** Eule evaluation result is negative and it is not firing.
- **Pending:** Rule evaluation result is positive and it will fire after the **for** period passes.
- **Firing:** Rule evaluation result is positive and it is spawning alerts. 

```

### Rule health

Rule health can be one of the following:

- **ok -** rule is being successfully evaluated.

> - **nodata -** rule is being evaluated, but no data is is returned. Depending on rule configuration this can result in either **Normal** or **Firing** state.

Reviewer 2: This is a little confusing. In the UI, we say `Alert state if no data or all values are null` above the configuration drop down. So, for example, we have a `Normal` rule with `nodata` health and a `NoData` instance state. 
![Alerting__Alert_rules_-_Grafana](https://user-images.githubusercontent.com/4461026/119571881-c424ec00-bd66-11eb-8815-fce7a2f4e8aa.png)

Author: Yeah, it is confusing :/ I think "nodata" instance only makes sense in the context of implementation details... Not sure we should be showing it at all

I think I'll do as Jita suggest I'll remove state explanations from this page, to be explained elsewhere in the docs

> - **ok -** rule is being successfully evaluated.
> - **nodata -** rule is being evaluated, but no data is is returned. Depending on rule configuration this can result in either **Normal** or **Firing** state.
> - **error -** attempting to evaluate the rule has resulted in error. Hover mouse over the **error** label to see error details in a popover.

Reviewer: ```suggestion
- **ok:** Rule is being successfully evaluated.
- **nodata:** Rule is being evaluated but no data is is returned. Depending on the rule configuration, this can result in either **Normal** or **Firing** state.
- **error:** Attempting to evaluate the rule has resulted in an error. Hover mouse over the **error** label to see the error details.
```

### Matching instances

> Matching instance are the alerts spawned by evaluating this rule. In case of grafana rules that use classic conditions there will be just one. In case of multi dimensional Grafana rules, Prometheus or Loki rules there can be several. Labels for an alert instance consist of rule labels, labels that come from evaluation result as well as the **alertname** label that matches the name of the rule.  

Reviewer: ```suggestion
Matching instances are the alerts spawned by evaluating this rule. In the case of unified rules that use classic conditions, there will be just one. For multi dimensional rules, Prometheus, or Loki rules there can be several. Labels for an alert instance consist of rule labels, labels that come from evaluation results as well as the **alertname** label that matches the name of the rule.  
```

Reviewer 2: Matching instances are the alerts that result from the evaluation of this rule, identified by their unique label set. If a rule uses a classic condition, there can only be one matching instance. For multi-dimensional, Prometheus, or Loki rules, there can be multiple instances. The label set for an alert instance consist of alerting rule labels, the labels produced by evaluation results to identify unique time series, and the **alertname** label that matches the title of the alerting rule. In the case of duplicate label keys for time series and alerting rule, the value from the alerting rule will take precedence.

Author: this is great stuff @davidmparrott ! However at Jita's suggestion I removed conceptual info from this page. I think we'll be able to link to architectural info instead

### Edit or delete rule

> To edit or delete a rule, expand this rule to reveal rule controls. Click "Edit" button to go to rule editing form. Click "Delete" button to delete a rule. Grafana rules can only be edited or deleted by users with Edit permissions for the folder which contains the rule. Prometheus or Loki rules can be edited or deleted by user with Editor or Admin role. 

Reviewer: ```suggestion

Grafana rules can only be edited or deleted by users with Edit permissions for the folder which contains the rule. Prometheus or Loki rules can be edited or deleted by users with Editor or Admin roles. 

To edit or delete a rule:

1. Expand this rule to reveal rule controls. 
1. Click **Edit** to go to the rule editing form. Make changes using instruction in ??
1. Click **Delete"** to delete a rule. 
```
